lib/test.js

```diff
@@ -0,0 +1,179 @@
+/*!
+ * Copyright (c) 2012-2019 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const bedrock = require('bedrock');
+const {config, util: {BedrockError}} = bedrock;
+const path = require('path');
+
+// module API
+const api = {};
+module.exports = api;
+
+bedrock.events.on('bedrock-cli.init', function(callback) {
+  // add test command
+  const command = bedrock.program
+    .command('test')
+    .description('run tests')
+    .option(
+      '--framework <frameworks>',
+      'A set of comma-delimited test frameworks to run. [all] ' +
+      '(' + config.test.frameworks.join(', ') + ')')
+    .action(function() {
+      config.cli.command = command;
+    });
+
+  // allow special test configs to load
+  bedrock.events.emit('bedrock-cli.test.configure', command, callback);
+});
+
+bedrock.events.on('bedrock-cli.ready', function(callback) {
+  const command = config.cli.command;
+  if(command.name() !== 'test') {
+    return callback();
+  }
+
+  // set reporter
+  if(command.mochaReporter) {
+    config.mocha.options.reporter = command.mochaReporter;
+  }
+
+  bedrock.events.emit('bedrock.test.configure', callback);
```

Comment:
Is it supposed to be the case that this event cannot be canceled without also canceling `bedrock-cli.ready`?

Side note: Ideally, we'd move this stuff over to async+await/promises, but that can wait.

Reply:
@dlongley this is same behavior we've always had here afaik, do you see some difficulty with it?

Reply:
It seems like it could be a problem if someone cancels `bedrock.test.configure` ... this has probably been a long standing issue. Let's ignore in this PR, I'll file an issue against it.